### PR TITLE
config: bump hibernate search to jdk 17

### PR DESCRIPTION
**Pull request by romani**
Master build is failing 

They migrated https://github.com/hibernate/hibernate-search/commit/81a280863aab3e42a8f9eb52d3c277b5c7731373

### CODE REVIEW

**Comment by romani** [comment]
>E: Unable to locate package adoptopenjdk-17-hotspot

We need to move it out of Travis.

**Comment by pbludov** [comment]
From https://adoptopenjdk.net/installation.html#
> 24th July 2021: [AdoptOpenJDK is moving to the Eclipse Foundation](https://blog.adoptopenjdk.net/2021/03/transition-to-eclipse-an-update/) and rebranding.
Our July 2021 and future releases will come from [Adoptium.net](https://www.adoptium.net/)

We need to switch to adoptium builds somehow.

**Comment by pbludov** [comment]
We can use https://sdkman.io/jdks#tem to install JDK

**Comment by romani** [comment]
We are on special architecture, general solutions unlikely to work for us.
I made new arch to work for us only with huge help from support team.

**Comment by romani** [comment]
I will keep this PR open as reminder to fix Travis, if somebody open good PR to fix Travis, please close this PR

**Comment by pbludov** [comment]
sdkman Failed with

```bash
Looking for a previous installation of SDKMAN...

Looking for unzip...

Looking for zip...

Not found.

======================================================================================================

 Please install zip on your system using your favourite package manager.

 Restart after installing zip.

======================================================================================================

The command "./.ci/travis.sh install-jdk-17" failed and exited with 1 during .

```

We should install unzip as well.

**Comment by pbludov** [comment]
sdkman installed, but failed to recognize ppc64le platform:
```bash
================================================================================

Available Java Versions for Exotic

================================================================================

 Vendor        | Use | Version      | Dist    | Status     | Identifier

--------------------------------------------------------------------------------

No versions available for your platform at this time.

================================================================================

Omit Identifier to install default version 17.0.2-tem:

    $ sdk install java

Use TAB completion to discover available versions

    $ sdk install java [TAB]

Or install a specific version by Identifier:

    $ sdk install java 17.0.2-tem
```

**Comment by pbludov** [comment]
Related sdkman issue https://github.com/sdkman/sdkman-cli/issues/1038

**Comment by romani** [comment]
Much easier to move haearch out of Travis

File: .travis.yml [hunk 1]
```diff
@@ -59,9 +60,9 @@ jobs:
         - CMD="$CMD1 && $CMD2"
         - USE_MAVEN_REPO="true"
 
-    - jdk: openjdk11
-      env:
+    - env:
         - DESC="NoErrorTest - Hibernate Search"
+        - CUSTOM_JDK_VERSION="temurin-17-jdk"
```
Comment (romani) [inline .travis.yml:65]: Let's add adoptium to name as it is custom jdk from certain provider.

**Review by romani — COMMENTED** [review]
Can you share link from where you get this  approach.

File: .travis.yml (hunk 1 above)
Comment (pbludov) [reply .travis.yml]: done

**Comment by pbludov** [comment]
> Can you share link from where you get this approach ?

It was copied from the same file, as it was done for `CUSTOM_MVN_VERSION`

**Comment by pbludov** [comment]
The commands were found here: https://adoptium.net/installation/linux/

**Review by romani — COMMENTED** [review]
Looks good

**Comment by romani** [comment]
List of all platforms https://adoptium.net/supported-platforms/

Our platform might match "Linux (PowerPC 64-bit Little Endian)"

https://en.m.wikipedia.org/wiki/Ppc64

**Comment by romani** [comment]
@rnveach , @nick-mancuso , @strkkk , FYI

**Pull request merged**